---
id: attr-no-duplication
title: attr-no-duplication
description: Prevents duplicate attributes in a single HTML element to ensure valid and clean markup.
---

import { Badge } from '@astrojs/starlight/components';

The same attribute can't be specified twice.

Level: <Badge text="Error" variant="danger" />

## Config value

1. true: enable rule
2. false: disable rule

### The following patterns are **not** considered rule violations:

```html
<img src="a.png" />
```

### The following pattern is considered a rule violation:

```html
<img src="a.png" src="b.png" />
```
