---
id: doctype-html5
title: doctype-html5
description: Ensures the doctype is HTML5.
---

import { Badge } from '@astrojs/starlight/components';

Doctype must be HTML5.

Level: <Badge text="Warning" variant="caution" />

## Config value

- `true`: enable rule
- `false`: disable rule

### The following pattern is **not** considered a rule violations

```html
<!DOCTYPE html>
<html></html>
```

### The following pattern is considered a rule violation:

```html
<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01//EN" "https://www.w3.org/TR/html4/strict.dtd">
<html></html>
```

## Why this rule is important

The DOCTYPE declaration is required for HTML5 compliance and to ensure that the document is parsed correctly.

Further reading: [MDN Web Docs - Doctype](https://developer.mozilla.org/en-US/docs/Glossary/Doctype)
