---
id: doctype-first
title: doctype-first
description: Ensures the doctype declaration appears before any non-comment content in HTML documents.
---

import { Badge } from '@astrojs/starlight/components';

Doctype must be declared before any non-comment content. Comments and whitespace are allowed before the DOCTYPE declaration.

Level: <Badge text="Error" variant="danger" />

## Config value

- `true`: enable rule
- `false`: disable rule

```html
<!-- Comment before doctype -->
<!DOCTYPE html>
<html></html>
```

### The following patterns are considered rule violations

```html
<html></html>
```

```html
<div>Content before doctype</div>
<!DOCTYPE html>
<html></html>
```

### Why this rule is important

The DOCTYPE declaration is required for HTML5 compliance and to ensure that the document is parsed correctly.
