---
id: space-tab-mixed-disabled
title: space-tab-mixed-disabled
description: Prevents mixing spaces and tabs for indentation at the beginning of lines to ensure consistent formatting.
---

import { Badge } from '@astrojs/starlight/components';

Blank-space (space and tab) characters should not be mixed in the beginning of any line.

Level: <Badge text="Warning" variant="caution" />

## Config values

- `space`: space mode (only spaces are valid for indentation)
- `space4`: space mode and require 4 space characters
- `tab`: tab mode (only tab characters are valid for indentation)
- `false`: disable rule

### Example

```json
{
  ...
  "space-tab-mixed-disabled": space4
  ...
}
```

### The following patterns are **not** considered rule violations

```html
      <img src="tab.png" />
33333333<img src="space.png" />
```

### The following pattern is considered a rule violation:

```html
   333<img src="tab_before_space.png" />
3333   <img src="space_before_tab.png" />
```

:::note
In the examples above, spaces and tabs are represented by `3` and ``, respectively, to make the difference visible.
:::
